tendrl/commons/objects/node_context/__init__.py

```diff
@@ -111,10 +111,16 @@ def on_change(self, attr, prev_value, current_value):
                     "WARNING",
                     node_id=self.node_id
                 )
-
                 _tc = NS.tendrl.objects.TendrlContext(
                     node_id=self.node_id
                 ).load()
+                # Load cluster_node_context will load node_context
+                # and it will be updated with latest values
+                cluster_node_context = NS.tendrl.objects.ClusterNodeContext(
+                    node_id=self.node_id,
+                    integration_id=_tc.integration_id
+                )
```

Comment:
Every time on_change is called you will save a new cluster_node_context?  Please "load" this object instead of making a new one every time

cluster_node_context = NS.tendrl.objects.ClusterNodeContext(
                    node_id=self.node_id,
                    integration_id=_tc.integration_id
        ).load()

Reply:
Load cannot be done , because we need to update the cluster node context which internally loads node context. If we do a load , the cluster node context's few fields get overridden by the old fields which fails the purpose of this change. What we can do is , create the object -> Save the object(Details populate to etcd) -> Delete the object.